Config/editeurTerrain.py

```diff
@@ -0,0 +1,243 @@
+                (x + 1) * cell_size + offset, (y + 1) * cell_size + offset,
+                fill=cell_color, outline="gray"
+            )
+    if current_point is not None:
+
+        create_circle(canvas, current_point[0][0], current_point[0][1],cell_size / 10)
+    directions = ["E","W","N","S"]
+    offset_dict = {
+        "E": (0.5 * cell_size, 0),
+        "W": (-0.5 * cell_size, 0),
+        "N": (0,-0.5 * cell_size),
+        "S": (0, 0.5 * cell_size)
+    }
+    for x in range(num_cells_x):
+        for y in range(num_cells_y):
+            for i1 in range(len(directions)):
+                dir1 = directions[i1]
+                if graph[x][y].getValue(dir1) == 0:
+                    continue
+                for i2 in range(i1+1, len(directions)):
+                    dir2 = directions[i2]
+                    if graph[x][y].getValue(dir2) == 0:
+                        continue
+                    pos_center = ((x + 0.5) * cell_size, (y + 0.5) * cell_size)
+                    if (dir1 in ["N","S"] and dir2 in ["N","S"]) or (dir1 in ["W","E"] and dir2 in ["W","E"]):
+                        canvas.create_line(pos_center[0] + offset_dict[dir1][0] + offset, pos_center[1] + offset_dict[dir1][1] + offset,
+                                   pos_center[0] + offset_dict[dir2][0] + offset, pos_center[1] + offset_dict[dir2][1] + offset, fill="black", width=5)
+                    else:
+                        if dir1 + dir2 in ["WS","SW"]:
+                            start = 0
+                            extent = 90
+                        elif dir1 + dir2 in ["WN","NW"]:
+                            start = -90
+                            extent = 90
+                        elif dir1 + dir2 in ["ES","SE"]:
+                            start = 90
+                            extent = 90
+                        elif dir1 + dir2 in ["EN","NE"]:
+                            start = 180
+                            extent = 90
+                        canvas.create_arc(pos_center[0] + 2*offset_dict[dir1][0] + offset, pos_center[1] + 2*offset_dict[dir1][1] + offset,
+                                   pos_center[0] + 2*offset_dict[dir2][0] + offset, pos_center[1] + 2*offset_dict[dir2][1] + offset, start=start, extent=extent, style=tk.ARC, width=5)
+    saveToFile(graph, file_name)
+horizontal_points = define_horizontal_points()
+vertical_points = define_vertical_points()
+
+# Initialize the GUI
+root = tk.Tk()
+root.title("Grid")
+
+
+# Calculate the size of the canvas based on the number of cells and cell size
+canvas_width = num_cells_x * cell_size + 2 * offset
+canvas_height = num_cells_y * cell_size + 2 * offset
+
+# Create a canvas to draw the grid
+canvas = tk.Canvas(root, width=canvas_width, height=canvas_height)
+canvas.pack()
+
+# Bind the click event to the canvas
+canvas.bind("<Button-1>", handle_click)
+
+# Draw the initial grid
+draw_grid()
+
+# Start the GUI event loop
+root.mainloop()
```

Comment:
Needs an extra newline at the end of the file.

Reply:
Corrected in commit a1338f97a13522b73b911a81b94fcef0305ff02d